### PR TITLE
Added: Bento support for Splitting AI

### PR DESCRIPTION
**Pull request by iftwigs**
Adding support for AI containerization for File Splitting, testing of a new submodule and documentation for it and other containerized AIs' pipeline.

### CODE REVIEW

File: konfuzio_sdk/bento/file_splitting/schemas.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+"""Define pydantic models for request and response from the Splitting AI."""
```
Comment (iftwigs) [inline konfuzio_sdk/bento/file_splitting/schemas.py:1]: todo add it to packages

File: konfuzio_sdk/bento/file_splitting/schemas.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    text: str
+    bboxes: Optional[Dict[int, Bbox20240930]]
+    pages: Optional[List[SplitRequest20240930Page]]
+
+
+class SplitResponse20240930(BaseModel):
+    """Describe a scheme for the splitting response on 30/09/2024."""
+
+    class Subdocument(BaseModel):
```
Comment (iftwigs) [inline konfuzio_sdk/bento/file_splitting/schemas.py:44]: rename to Splitting results

File: konfuzio_sdk/bento/file_splitting/schemas.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+            category_id: int
+            confidence: float
+            category_name: str
+
+        page_ids: List[PageId]
+        category: int
+        categories: List[CategoryAnnotation]
+
+    subdocuments: List[Subdocument]
```
Comment (iftwigs) [inline konfuzio_sdk/bento/file_splitting/schemas.py:63]: rename to splitting results

File: konfuzio_sdk/bento/file_splitting/schemas.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            page_id: int
+
+        class CategoryAnnotation(BaseModel):
+            """Describe a scheme for a CategoryAnnotation on 30/09/2024."""
+
+            category_id: int
+            confidence: float
+            category_name: str
+
+        page_ids: List[PageId]
```
Comment (iftwigs) [inline konfuzio_sdk/bento/file_splitting/schemas.py:59]: list[int]

File: konfuzio_sdk/bento/file_splitting/splittingai_service.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+# load ai model name from AI_MODEL_NAME file in parent directory
+ai_model_name_file = os.path.join(os.path.dirname(os.path.dirname(__file__)), 'AI_MODEL_NAME')
+ai_model_name = open(ai_model_name_file).read().strip()
+
+app = FastAPI()
+
+
+@bentoml.service(
+    traffic={
+        'timeout': 3600,  # Hard limit for file splitting calls is 1 hour
```
Comment (iftwigs) [inline konfuzio_sdk/bento/file_splitting/splittingai_service.py:31]: set as high as possible

File: konfuzio_sdk/bento/file_splitting/splittingai_service.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        # Don't process more than 2 documents at a time. Will respond with 429 if more come.
+        # Clients should implement a retry strategy for 429.
+        # Servers should implement a scaling strategy and start multiple services when high load is present.
+        'max_concurrency': 2,
```
Comment (iftwigs) [inline konfuzio_sdk/bento/file_splitting/splittingai_service.py:35]: use env variable configured on the server side instead of hardcoding?

Comment (zypriafl) [reply konfuzio_sdk/bento/file_splitting/splittingai_service.py:35]: Please do like in :
https://github.com/konfuzio-ai/konfuzio-sdk/blob/a4d8d04c6a4041c00ffd9d5c6a6338da20dd1f02/konfuzio_sdk/bento/extraction/rfextractionai_service.py#L31

File: konfuzio_sdk/bento/file_splitting/splittingai_service.py [hunk 7]
```diff
@@ -28,7 +28,7 @@
 
 @bentoml.service(
     traffic={
-        'timeout': 9999,
+        'timeout': int(os.environ.get('BENTO_SERVICE_TIMEOUT', '3600')),
         # Don't process more than 2 documents at a time. Will respond with 429 if more come.
         # Clients should implement a retry strategy for 429.
         # Servers should implement a scaling strategy and start multiple services when high load is present.
```
Comment (zypriafl) [inline konfuzio_sdk/bento/file_splitting/splittingai_service.py:34]: Concurrency should also be configureable.
https://github.com/konfuzio-ai/konfuzio-sdk/blob/a4d8d04c6a4041c00ffd9d5c6a6338da20dd1f02/konfuzio_sdk/bento/extraction/rfextractionai_service.py#L35C1-L36C1

**Pull request merged**